src/Facebook/InstantArticles/Transformer/Rules/Rule.php

```diff
@@ -53,20 +39,20 @@ public static function create(): Rule
         );
     }
 
-    public static function createFrom(array<string, mixed> $configuration): Rule
+    public static function createFrom(dict<string, mixed> $configuration): Rule
     {
         throw new \Exception(
             'All Rule class extensions should implement the '.
             'Rule::createFrom($configuration) method'
         );
     }
 
-    public static function retrieveProperty(array<string, mixed> $properties, string $property_name): ?array<string, mixed>
+    public static function retrieveProperty(dict<string, mixed> $properties, string $property_name): ?dict<string, mixed>
     {
         if (array_key_exists('properties', $properties)) {
             $mappedProperties = $properties['properties'];
             if (is_array($mappedProperties) && array_key_exists($property_name, $mappedProperties)) {
-                return $mappedProperties[$property_name];
+                return dict($mappedProperties[$property_name]);
```

Comment:
What if `$mappedProperties[$property_name]` is not an array (bogus rules JSON)?

Example:

```json
{ "rules":[{"class":"ImageRule", "selector": "div.image", "properties": "clowntown"}] }
```

Reply:
This is all fine, added few more Test cases that raises no error while processing.